source/utilities.tex

```diff
@@ -6463,6 +6464,2036 @@
+\item
+\tcode{is_constructible_v<T, expected<U, G>\&>} is \tcode{false}; and
+\item
+\tcode{is_constructible_v<T, expected<U, G>>} is \tcode{false}; and
+\item
+\tcode{is_constructible_v<T, const expected<U, G>\&>} is \tcode{false}; and
+\item
+\tcode{is_constructible_v<T, const expected<U, G>>} is \tcode{false}; and
+\item
+\tcode{is_convertible_v<expected<U, G>\&, T>} is \tcode{false}; and
+\item
+\tcode{is_convertible_v<expected<U, G>\&\&, T>} is \tcode{false}; and
+\item
+\tcode{is_convertible_v<const expected<U, G>\&, T>} is \tcode{false}; and
+\item
+\tcode{is_convertible_v<const expected<U, G>\&\&, T>} is \tcode{false}; and
+\item
+\tcode{is_constructible_v<unexpected<E>, expected<U, G>\&>} is \tcode{false}; and
+\item
+\tcode{is_constructible_v<unexpected<E>, expected<U, G>>} is \tcode{false}; and
+\item
+\tcode{is_constructible_v<unexpected<E>, const expected<U, G>\&>} is \tcode{false}; and
+\item
+\tcode{is_constructible_v<unexpected<E>, const expected<U, G>>} is \tcode{false}.
+\end{itemize}
+
+\pnum
+\effects
+If \tcode{rhs.has_value()},
+direct-non-list-initializes \exposid{val} with \tcode{std::forward<UF>(*rhs)}.
+Otherwise,
+direct-non-list-initializes \exposid{unex} with \tcode{std::forward<GF>(rhs.error())}.
+
+\pnum
+\ensures
+\tcode{rhs.has_value()} is unchanged;
+\tcode{rhs.has_value() == this->has_value()} is \tcode{true}.
+
+\pnum
+\throws
+Any exception thrown by the initialization of \exposid{val} or \exposid{unex}.
+
+\pnum
+\remarks
+The expression inside \tcode{explicit} is equivalent to
+\tcode{!is_convertible_v<UF, T> || !is_convertible_v<GF, E>}.
+\end{itemdescr}
+
+\indexlibraryctor{expected}%
+\begin{itemdecl}
+template<class U = T>
+  constexpr explicit(!is_convertible_v<U, T>) expected(U&& v);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constraints
+\begin{itemize}
+\item
+\tcode{is_same_v<remove_cvref_t<U>, in_place_t>} is \tcode{false}; and
+\item
+\tcode{is_same_v<expected<T, E>, remove_cvref_t<U>>} is \tcode{false}; and
+\item
+\tcode{remove_cvref_t<U>} is not a specialization of \tcode{unexpected}; and
+\item
+\tcode{is_constructible_v<T, U>} is \tcode{true}.
+\end{itemize}
+
+\pnum
+\effects
+Direct-non-list-initializes \exposid{val} with \tcode{std::forward<U>(v)}.
+
+\pnum
+\ensures
+\tcode{has_value()} is \tcode{true}.
+
+\pnum
+\throws
+Any exception thrown by the initialization of \exposid{val}.
+\end{itemdescr}
+
+\indexlibraryctor{expected}%
+\begin{itemdecl}
+template<class G>
+  constexpr explicit(!is_convertible_v<const G&, E>) expected(const unexpected<G>& e);
+template<class G>
+  constexpr explicit(!is_convertible_v<G, E>) expected(unexpected<G>&& e);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{GF} be \tcode{const G\&} for the first overload and
+\tcode{G} for the second overload.
+
+\pnum
+\constraints
+\tcode{is_constructible_v<E, GF>} is \tcode{true}.
+
+\pnum
+\effects
+Direct-non-list-initializes \exposid{unex} with \tcode{std::forward<GF>(e.error())}.
```

Comment:
```suggestion
Direct-non-list-initializes \exposid{unex} with \tcode{std::forward<GF>(e.value())}.
```

`unexpected` calls its member accessor `value()`, not `error()`.

Reply:
Actually, I'd expected `unexpected<E>`'s member accessor to be `error()` instead of `value()`... but I guess that should be a later LWG issue and is out of scope for this PR.

Reply:
Changing that would be totally out of scope for an LWG issue, because it's not a defect. It's the approved design.

It should be `error()` here though, good catch.

Reply:
Oh, I thought LWG issues can be deferred to LEWG to discuss design changes... Does this means that I have to write a proposal to change this?

Reply:
Yes. This is not a defect. If you don't like the design, write a paper for LEWG.

Reply:
Thank you, I think I'll write one in the weekend... BTW, the poll is saying R12, but I cannot find it anywhere ([R11 source github](https://github.com/jfbastien/papers) seems not updated yet). Is there a changelog of R12 that I can base on?